### ISSUE TITLE
Screen readers are noisy and unhelpful

### ISSUE THREAD

**Issue by turt2live** [vector-im/element-web#9747]
Many of these are duplicate of other issues, however I'm going for a checklist of things which can be solved:

* [x] Read receipts are read out by screen readers with no context - https://github.com/vector-im/riot-web/issues/2716 - Should we add context and put the speech behind a flag?
* [x] NVDA says "Riot Google Chrome Document Row 1 Column 1 Edit Send a message unencrypted"
* [x] Message action bar is read out by screen readers
* [x] Messages are read out loud 3 times: once when grey (sending), once when sent, and once more for good measure. The third one has a full timestamp read out ("May 16, 2019 at 10:25 PM")
* [x] Screen readers read the timeline from hours out of view, and often don't include the message bodies. For example: "4:32 PM [describes message action bar] 4:33 PM [describes message action bar] ..."
* [x] `+` next to people and rooms is just labelled as "Button" (woo, so helpful)
* [x] Down arrows sometimes say "Button Number Matrix HQ" instead of "Button Matrix HQ" - can we get it to say "Room Matrix HQ" or something?
* [x] Cannot distinguish between room headers and rooms when using arrow keys to go down the list ("Button People", "Button Rooms", "Button Matrix HQ")
* [x] Flair is noisy ("plus matrix matrix dot org matrix core team plus matrix matrix dot org")
* [x] "Files button collapsed expanded files panel" when clicking on files tab
* [x] "Complimentary landmark TravisR button" for clicking name at top left (as welcoming as it is to be my own personal landmark)
* [x] Settings dialogs do not annotate tabs, cannot move between them. No idea what tab I'm on - no verbal cue
* [ ] Device name settings doesn't make any sense with speech cues - "Desktop row 1 column 1 device table editable desktop selection negative 1 multiline editable desktop" (my device name is "Desktop") (https://github.com/vector-im/riot-web/issues/11288)
* [x] Switches in settings are not keyboard accessible (tab skips)
* [x] Setting tabs are not keyboard accessible
* [ ] Many links and buttons are just their names, eg "restore from backup button" - restore what backup? or "here button" for a "click here" link - click for what?
* [x] Riot breaks through NVDA's sleep mode - can't fix
* [x] Hovering messages causes readers to read out timestamps
* [x] Readers keep reading old rooms when you've already switched several times
* [ ] Implement https://github.com/matrix-org/matrix-react-sdk/pull/3019 proper
* [ ] Probably more...

TODO: Associate with individual issues if they already exist
TODO: Check login/registration etc

**Comment by t3chguy** [comment]
> "Files button collapsed expanded files panel" when clicking on files tab

is now "Files Tab 2 of 3" then "Selected"

**Comment by duxovni** [comment]
This issue is pretty old, and at least a couple of the outstanding sub-issues on the checklist seem possibly fixed or no longer relevant. I'll close this, and we can open new issues for screen reader accessibility issues that are still present, perhaps with #20099 as a starting point.
